src/main/kotlin/pt/up/fe/ni/website/backend/controller/AuthController.kt

```diff
@@ -28,10 +31,29 @@ class AuthController(val authService: AuthService) {
         return mapOf("access_token" to accessToken)
     }
 
+    @PreAuthorize("isAuthenticated()")
     @GetMapping
-    @PreAuthorize("hasRole('MEMBER')")
-    fun checkAuthentication(): Map<String, Account> {
+    fun checkAuthentication(): Map<String, Any> {
+        val authentication = SecurityContextHolder.getContext().authentication
         val account = authService.getAuthenticatedAccount()
-        return mapOf("authenticated_user" to account)
+        return mapOf(
+            "authenticated_user" to account,
+            "jwt_permissions" to authentication.authorities.map { it.toString() }.toList()
+        )
+    }
+
+    @PreAuthorize("@authService.hasPermission(#permission.trim().toUpperCase())")
+    @GetMapping("/hasPermission/{permission}")
+    fun protectedPermission(@PathVariable permission: String): Map<String, String> {
+        return mapOf("message" to "You have permission to access this endpoint!")
+    }
```

Comment:
Is this endpoint just an example? It'd be nice to apply the logic to the existing endpoints instead. If this is outside the PR's scope, please open an issue with high priority

Reply:
If it's meant to be used to check permissions, I think a 200 empty response is more RESTful

Reply:
The idea here was in fact to be just a temporary endpoint, to show how to use the annotation while there are not other endpoints that use this approach.